tests/cephfs/cephfs_mirroring/test_cephfs_mirroring_metrics.py

```diff
@@ -0,0 +1,664 @@
+            )
+        else:
+            log.error(
+                f"Snapshot Deleted count : {snap_count} doesn't exist in Metrics : {snaps_deleted}"
+            )
+            raise CommandFailed("Snapshot Deleted count is mismatching in Metrics")
+
+        log.info("Remove subvolume paths from mirroring")
+        fs_mirroring_utils.remove_path_from_mirroring(
+            source_clients[0], source_fs_new, subvol1_path
+        )
+        fs_mirroring_utils.remove_path_from_mirroring(
+            source_clients[0], source_fs_new, subvol2_path
+        )
+
+        log.info("Remove the newly added Filesystem from mirroring")
+        log.info(f"Disable cephfs mirroring module on the {source_fs_new}")
+        fs_mirroring_utils.disable_snapshot_mirroring(source_fs_new, source_clients[0])
+        time.sleep(20)
+
+        data_after_fs_remove = fs_mirroring_utils.get_cephfs_mirror_counters(
+            cephfs_mirror_node, fsid, asok_file
+        )
+        resource_name = "cephfs_mirror_peers"
+        filesystem_name = source_fs_new
+        label, counter = fs_mirroring_utils.get_labels_and_counters(
+            resource_name, filesystem_name, data_after_fs_remove
+        )
+
+        if label is None and counter is None:
+            log.info(f"FS : {source_fs_new} is removed from the metrics.")
+        else:
+            log.error(f"FS : {source_fs_new} still exists in the metrics")
+            raise CommandFailed("Mismatch in the metrics.")
+
+        resource_name = "cephfs_mirror_mirrored_filesystems"
+        filesystem_name = source_fs_new
+        label, counter = fs_mirroring_utils.get_labels_and_counters(
+            resource_name, filesystem_name, data_after_fs_remove
+        )
+        if label is None and counter is None:
+            log.info(f"Mirrored FS : {source_fs_new} is removed from the metrics.")
+        else:
+            log.error(f"Mirrored FS : {source_fs_new} still exists in the metrics")
+            raise CommandFailed("Mismatch in the metrics.")
+        return 0
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Clean up the system")
+        log.info("Unmount the paths on source")
+        mounting_dirs = [
+            kernel_mounting_dir_1,
+            fuse_mounting_dir_1,
+        ]
+        for mounting_dir in mounting_dirs:
+            source_clients[0].exec_command(sudo=True, cmd=f"umount -l {mounting_dir}")
+
+        log.info("Remove peer from snapshot mirroring")
+        mirroring_data = [
+            (source_fs_new, peer_uuid_new),
+        ]
+        for fs, peer_uuid_new in mirroring_data:
+            if fs_mirroring_utils.remove_snapshot_mirror_peer(
+                source_clients[0], fs, peer_uuid_new
+            ):
+                log.info("Peer removal is successful.")
+            else:
+                log.error("Peer removal failed.")
```

Comment:
As per Logs http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-J62CQY/Validate_the_CephFS_Mirroring_Metrics_0.log There is a failure here as "filesystem cephfs-snew is not mirrored" may be because we have already removed it from mirroring before cleanup at fs_mirroring_utils.remove_path_from_mirroring ?
So please handle this code accordingly.